src/main/java/bio/terra/service/load/flight/LoadLockStep.java

```diff
@@ -9,21 +9,24 @@
 import bio.terra.stairway.StepStatus;
 import java.util.UUID;
 
-// This step is meant to be shared by dataset and filesystem flights for locking the load tag.
-// It expects to find LoadMapKeys.LOAD_TAG in the parameters or working map.
-
 public class LoadLockStep implements Step {
   private final LoadService loadService;
+  private final UUID datasetId;
 
-  public LoadLockStep(LoadService loadService) {
+  /**
+   * This step is meant to be shared by dataset and filesystem flights for locking the load tag. It
+   * expects to find LoadMapKeys.LOAD_TAG in the parameters or working map.
+   */
+  public LoadLockStep(LoadService loadService, UUID datasetId) {
     this.loadService = loadService;
+    this.datasetId = datasetId;
```

Comment:
If reviewers prefer, I could instead add an `IngestUtils` method to pull the `datasetId` from the flight's input parameters, rather than pass it into step construction.  We are not consistent on approach here.

Reply:
That's what I was going to suggest, the dataset ID is always in `JobMapKeys.DATASET_ID` in these flights so I'd extract it in the step instead of passing it down via constructor.

Reply:
Updated!